### PR TITLE
Task 613 make default workspac

### PR DESCRIPTION
**Pull request by mjmeijer**
I think this will put the workspace location suggested on first launch in the proper place on each platform.

### CODE REVIEW

**Comment by jantje** [comment]
There is quite some change that I think is unneeded/bad.
I'll comment in the "Files changed".

File: io.sloeber.application/.classpath [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
 	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
 	<classpathentry kind="src" path="src"/>
-	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
-	<classpathentry kind="output" path="bin"/>
+	<classpathentry kind="output" path="target/classes"/>
```
Comment (jantje) [inline io.sloeber.application/.classpath]: Why do we want to change the output path? I'm not sure what con is about.

File: io.sloeber.product/arduino.product [hunk 2]
```diff
@@ -46,6 +46,7 @@ https://github.com/jantje/arduino-eclipse-plugin/graphs/contributors
    <launcher name="sloeber-ide">
       <linux icon="launch/icon.xpm"/>
       <macosx icon="launch/icon.icns"/>
+      <solaris/>
```
Comment (jantje) [inline io.sloeber.product/arduino.product]: I thought @wimjongman had removed this.

File: io.sloeber.product/arduino.product [hunk 3]
```diff
@@ -367,6 +368,56 @@ rights to a jury trial in any resulting litigation.&lt;/p&gt;
...
+      <repository location="http://download.eclipse.org/eclipse/updates/4.6" enabled="true" />
+      <repository location="http://download.eclipse.org/releases/neon" enabled="true" />
+      <repository location="http://eclipse.baeyens.it/update/nightly" enabled="false" />
+      <repository location="http://eclipse.baeyens.it/update/V4/nightly" enabled="true" />
+      <repository location="http://eclipse.baeyens.it/update/V4/stable" enabled="true" />
+      <repository location="http://download.eclipse.org/eclipse/updates/4.6" enabled="true" />
+      <repository location="http://download.eclipse.org/releases/neon" enabled="true" />
+      <repository location="http://eclipse.baeyens.it/update/nightly" enabled="false" />
+      <repository location="http://eclipse.baeyens.it/update/V4/nightly" enabled="true" />
+      <repository location="http://eclipse.baeyens.it/update/V4/stable" enabled="true" />
```
Comment (jantje) [inline io.sloeber.product/arduino.product]: The same update site is added over and over again.

File: io.sloeber.product/neon-cdt-nebula.target [hunk 4]
```diff
@@ -44,5 +40,24 @@
...
+<unit id="ilg.gnuarmeclipse.templates.ad.feature.group" version="1.1.3.201606210758"/>
+<unit id="ilg.gnuarmeclipse.templates.cortexm.feature.group" version="1.3.1.201606210758"/>
+<unit id="ilg.gnuarmeclipse.debug.gdbjtag.qemu.feature.group" version="3.1.1.201606210758"/>
+<repository location="http://sloeber.io/gnuarmeclipse/stable"/>
+</location>
+<location includeAllPlatforms="true" includeConfigurePhase="true" includeMode="planner" includeSource="true" type="InstallableUnit">
+<unit id="org.eclipse.nebula.widgets.oscilloscope.feature.feature.group" version="1.2.0.201612121953"/>
+<repository location="http://download.eclipse.org/nebula/snapshot"/>
+</location>
 </locations>
```
Comment (jantje) [inline io.sloeber.product/neon-cdt-nebula.target]: This file is a new file with a wierd name and when I open it in my workspace the location is full of error tags.
Give the name why is gnuarmeclipse in here?

File: io.sloeber.product/p2.inf [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+units.2.version=1.0.0
+units.2.provides.1.namespace=org.eclipse.equinox.p2.iu
+units.2.provides.1.name=sloeber.configuration.default
+units.2.provides.1.version=1.0.0
+units.2.filter=(!(osgi.os=win32))
+units.2.touchpoint.id=org.eclipse.equinox.p2.osgi
+units.2.touchpoint.version=1.0.0
+units.2.instructions.configure=setProgramProperty(propName:osgi.instance.area.default,propValue:@user.home/Documents/sloeber-workspace);
+units.2.instructions.unconfigure=setProgramProperty(propName:osgi.instance.area.default,propValue:);
+
```
Comment (jantje) [inline io.sloeber.product/p2.inf]: This is a new file.
Can you explain me what it does and why it is needed?

File: io.sloeber.product/neon-cdt-nebula.target (hunk 4 above)
Comment (mjmeijer) [inline io.sloeber.product/neon-cdt-nebula.target]: http://www.vogella.com/tutorials/EclipseTargetPlatform/article.html

Define all plugins needed for compilation as separate from your working Eclipse with JDT and PDE

File: io.sloeber.product/p2.inf (hunk 5 above)
Comment (mjmeijer) [inline io.sloeber.product/p2.inf]: https://wiki.eclipse.org/Equinox/p2/Customizing_Metadata
and copied from http://bit.ly/2j04DvN
Makes the build create extra parameters in config.ini that define default workspace location

File: io.sloeber.product/neon-cdt-nebula.target (hunk 4 above)
Comment (jantje) [inline io.sloeber.product/neon-cdt-nebula.target]: any idea why I see all these errors? This is the previous version so maybe this fixes these errors ;-)
errors

File: io.sloeber.product/p2.inf (hunk 5 above)
Comment (jantje) [inline io.sloeber.product/p2.inf]: ok

**Pull request merged**